src/analyzer/protocol/quic/QUIC.spicy

```diff
@@ -453,58 +529,63 @@ type Packet = unit(from_client: bool, context: ConnectionIDInfo&) {
         # If we see a retry packet from the responder, reset the decryption
         # context such that the next DCID from the client is used for decryption.
         if ( self.long_header.is_retry ) {
-          context.client_initial_processed = False;
-          context.server_initial_processed = False;
-          context.initial_destination_conn_id = b"";
 
-          # Allow re-opening the SSL analyzer the next time around.
+          # Recreate all the crypto state on the next %init of Packet.
           zeek::protocol_handle_close(context.ssl_handle);
           unset context.ssl_handle;
+          context.client_crypto = Null;
+          context.server_crypto = Null;
+          context.client_sink = Null;
+          context.server_sink = Null;
+          self.crypto = Null;
+          self.crypto_sink = Null;
+
+          # Reset crypto state!
+          context.initial_destination_conn_id = Null;
```

Comment:
Nit: we could move this to a dedicated function, and call it from a new `on %error` impl. We don't need this right now since we do not perform resynchronization, but it might still make sense since we still have all the context swapped in.